### PR TITLE
3.4 Backports

### PR DESCRIPTION
**Pull request by nyalldawson**
Misc backports

### CODE REVIEW

File: src/gui/qgsgui.h [hunk 1]
```diff
@@ -41,10 +41,22 @@ class QgsWindowManagerInterface;
  * related to GUI classes.
  * \since QGIS 3.0
  */
-class GUI_EXPORT QgsGui
+class GUI_EXPORT QgsGui : public QObject
 {
+    Q_OBJECT
```
Comment (m-kuhn) [inline src/gui/qgsgui.h]: Is using `Q_GADGET` instead of `Q_OBJECT` possible here?

Comment (nyalldawson) [reply src/gui/qgsgui.h]: Possibly, but I figured sooner or later we'd need this to be a Q_OBJECT and want to emit a signal

Comment (m-kuhn) [reply src/gui/qgsgui.h]: I normally try to add overhead when it's needed and not prophylactic. Up to you.

**Review by m-kuhn — APPROVED** [review]

File: src/gui/qgsgui.h (hunk 1 above)
Comment (nyalldawson) [reply src/gui/qgsgui.h]: It's a singleton anyway. loosen up and have a :beer: !

Comment (m-kuhn) [reply src/gui/qgsgui.h]: Alright :+1: 
I just had a couple of days off, back to :milk_glass: now :)

**Comment by m-kuhn** [comment]
Thanks for the fix!

**Pull request merged**